docker/docker_client.go

```diff
@@ -998,20 +998,25 @@ func (c *dockerClient) getExternalBlob(ctx context.Context, urls []string) (io.R
 			resp.Body.Close()
 			continue
 		}
-		return resp.Body, getBlobSize(resp), nil
+
+		size, err := getBlobSize(resp)
+		if err != nil {
+			size = -1
+		}
+		return resp.Body, size, nil
 	}
 	if remoteErrors == nil {
 		return nil, 0, nil // fallback to non-external blob
 	}
 	return nil, 0, fmt.Errorf("failed fetching external blob from all urls: %w", multierr.Format("", ", ", "", remoteErrors))
 }
 
-func getBlobSize(resp *http.Response) int64 {
+func getBlobSize(resp *http.Response) (int64, error) {
 	size, err := strconv.ParseInt(resp.Header.Get("Content-Length"), 10, 64)
```

Comment:
This should report whether the header is missing, or present and invalid (via `resp.Header.Values`)